internal/pkg/cli/pipeline_list.go

```diff
@@ -75,60 +115,162 @@ func (o *listPipelineOpts) Ask() error {
 		}
 		o.appName = app
 	}
+
 	return nil
 }
 
 // Execute writes the pipelines.
 func (o *listPipelineOpts) Execute() error {
-	var out string
-	pipelines, err := o.pipelineLister.ListDeployedPipelines(o.appName)
+	ctx, cancel := context.WithTimeout(context.Background(), pipelineListTimeout)
+	defer cancel()
+
+	switch {
+	case o.shouldShowLocalPipelines && o.shouldOutputJSON:
+		return o.jsonOutputLocal(ctx)
+	case o.shouldShowLocalPipelines:
+		return o.humanOutputLocal()
+	case o.shouldOutputJSON:
+		return o.jsonOutputDeployed(ctx)
+	}
+
+	return o.humanOutputDeployed()
+}
+
+// jsonOutputLocal prints data about all pipelines in the current workspace.
+// If a local pipeline has been deployed, data from codepipeline is included.
+func (o *listPipelineOpts) jsonOutputLocal(ctx context.Context) error {
+	local, err := o.workspace.ListPipelines()
 	if err != nil {
-		return fmt.Errorf("list deployed pipelines in application %s: %w", o.appName, err)
+		return err
 	}
-	if o.shouldOutputJSON {
-		var pipelineInfo []*codepipeline.Pipeline
-		for _, pipeline := range pipelines {
-			info, err := o.codepipeline.GetPipeline(pipeline.ResourceName)
-			if err != nil {
-				return fmt.Errorf("get pipeline info for %s: %w", pipeline.Name, err)
-			}
-			pipelineInfo = append(pipelineInfo, info)
-		}
 
-		data, err := o.jsonOutput(pipelineInfo)
-		if err != nil {
-			return err
-		}
-		out = data
-	} else {
-		var pipelineNames []string
-		for _, pipeline := range pipelines {
-			pipelineNames = append(pipelineNames, pipeline.Name)
-		}
-		out = o.humanOutput(pipelineNames)
+	deployed, err := getDeployedPipelines(ctx, o.appName, o.pipelineLister, o.newDescriber)
+	if err != nil {
+		return err
+	}
+
+	cp := make(map[string]*describe.Pipeline)
+	for _, pipeline := range deployed {
+		cp[pipeline.Name] = pipeline
+	}
+
+	type combinedInfo struct {
+		Name         string `json:"name"`
+		ManifestPath string `json:"manifestPath"`
+		*describe.Pipeline
```

Comment:
`describe.Pipeline` also has a field called `Name` which marshals to `name` 🤔  Does this render two `name` fields in the JSON output?

Reply:
no - it will only marshal this `Name` field. If we removed this `Name` field and instantiated a `&describe.Pipeline{}` for the local-only pipelines, they would get a bunch of empty fields in their json output (like `region`, `accountId`, etc)).

Reply:
What if a local pipeline is deployed, could we have a `combinedInfo` thats like
```go
combinedInfo {
  Name: "myPipeline",
  manifestPath: "copilot/pipelines/myPipeline/manifest.yml`,
  Pipeline: &describe.Pipeline{
      Name: "myPipeline",
      Pipeline: codepipeline.Pipeline {
           Name: "pipeline-myApp-myPipeline-RANDOMSTRING",
       }
  },
}
```

Reply:
I wonder if we display too much info for `ls`? Ideally `ls` users only care about the list, otherwise they could use `show` 🤔

Reply:
input:
```
cli.combinedInfo{Name:"mypipeline", ManifestPath:"copilot/pipelines/mypipeline/manifest.yml", Pipeline:&describe.Pipeline{Name:"mypipeline", Pipeline:codepipeline.Pipeline{Name:"pipeline-myApp-myPipeline-RANDOMSTRING", Region:"", AccountID:"", Stages:[]*codepipeline.Stage(nil), CreatedAt:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), UpdatedAt:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC)}, Resources:[]*stack.Resource(nil)}}

```
json:
```
{"pipelines":[{"name":"mypipeline","manifestPath":"copilot/pipelines/mypipeline/manifest.yml","pipelineName":"pipeline-myApp-myPipeline-RANDOMSTRING","region":"","accountId":"","stages":null,"createdAt":"0001-01-01T00:00:00Z","updatedAt":"0001-01-01T00:00:00Z"}]}
```

Reply:
hmm interesting it seems like only one of `combinedInfo.Name` and `describe.Pipeline.Name` is getting marshaled into `name`. I didn't know this 😮

Reply:
What do folks think of
```json
{
      "name":"mypipeline",
      "manifestPath":"copilot/pipelines/mypipeline/manifest.yml",
      "pipelineName":"pipeline-myApp-myPipeline-RANDOMSTRING",
}
```
Would this be confusing? `pipelineName` could be empty but the blast radius is reduced since the pipeline is not deployed.

Reply:
discussed offline. i'll go ahead with @iamhopaul123's suggestion since we can always add in additional fields to `--json --local` in the future.